python/rmm/_lib/device_buffer.pyx

```diff
@@ -397,6 +377,8 @@ cpdef void copy_ptr_to_host(uintptr_t db,
         _copy_async(<const void*>db, <void*>&hb[0], len(hb),
                     cudaMemcpyDeviceToHost, stream.view())
 
+    if stream.c_is_default():
+        stream.c_synchronize()
 
 @cython.boundscheck(False)
 cpdef void copy_host_to_ptr(const unsigned char[::1] hb,
```

Comment:
We need to synchronize in this function as well. If we don't then the host buffer could be mutated or go out of scope before the copy finishes, i.e. `copy_host_to_ptr(np.asarray([1, 2, 3]), device_ptr)`

Reply:
Agreed, this is now done in https://github.com/rapidsai/rmm/pull/650/commits/64d279ba95841ac2d674209a037b60bd7df60c19 .